### PR TITLE
Improve webhook verification

### PR DESCRIPTION
**Pull request by zhouhao3**
Part of #855 

- Add validations for ImageUrl
- Add validations for Image.DiskFormat

### CODE REVIEW

**Comment by zhouhao3** [comment]
@elfosardo @dtantsur PTAL

**Comment by zhouhao3** [comment]
@furkatgofurov7 @fmuyassarov PTAL, thanks!

**Comment by zhouhao3** [comment]
@dtantsur @fmuyassarov @elfosardo This PR has been stuck for a long time, please take a look if you have time. Thanks.

File: apis/metal3.io/v1alpha1/baremetalhost_validation.go [hunk 1]
```diff
@@ -96,6 +103,10 @@ func validateBMCAccess(s BareMetalHostSpec, bmcAccess bmc.AccessDetails) []error
 		errs = append(errs, fmt.Errorf("BMC driver %s does not support secure boot", bmcAccess.Type()))
 	}
 
+	if s.Image != nil && s.Image.DiskFormat != nil && *s.Image.DiskFormat == "live-iso" && !bmcAccess.SupportsISOPreprovisioningImage() {
```
Comment (fmuyassarov) [inline apis/metal3.io/v1alpha1/baremetalhost_validation.go:106]: just a nit, maybe for the consistency we could define a constant for "live-iso" in https://github.com/metal3-io/baremetal-operator/blob/6966480dece8aed3cbf1f14180406548f87ec796/apis/metal3.io/v1alpha1/baremetalhost_types.go#L99

**Review by fmuyassarov — COMMENTED** [review]
/approve

File: apis/metal3.io/v1alpha1/baremetalhost_validation.go (hunk 1 above)
Comment (zhouhao3) [reply apis/metal3.io/v1alpha1/baremetalhost_validation.go]: updated, thanks.

**Comment by zhouhao3** [comment]
@honza @furkatgofurov7 PTAL

File: apis/metal3.io/v1alpha1/baremetalhost_validation_test.go [hunk 2]
```diff
@@ -437,6 +438,43 @@ func TestValidateCreate(t *testing.T) {
...
+					Image: &Image{
+						URL: "test1",
+					},
+				},
+			},
+			oldBMH:    nil,
+			wantedErr: "Image URL test1 is an invalid URL",
+		},
+		{
+			name: "Invalidate image URL",
```
Comment (furkatgofurov7) [inline apis/metal3.io/v1alpha1/baremetalhost_validation_test.go:460]: Do we need to update this as per `wantedErr` description?

Comment (zhouhao3) [reply apis/metal3.io/v1alpha1/baremetalhost_validation_test.go]: updated, thanks.

File: apis/metal3.io/v1alpha1/baremetalhost_validation_test.go [hunk 3]
```diff
@@ -437,6 +438,43 @@ func TestValidateCreate(t *testing.T) {
 			oldBMH:    nil,
 			wantedErr: "BMO validation: failed to parse BMC address information: BMC address hostname/IP : [fe80::fc33:62ff:fe33:8xff] is invalid",
 		},
+		{
+			name: "invalidateImageURL",
```
Comment (furkatgofurov7) [inline apis/metal3.io/v1alpha1/baremetalhost_validation_test.go:442]:
```suggestion
			name: "invalidImageURL",
```

File: apis/metal3.io/v1alpha1/baremetalhost_validation_test.go [hunk 4]
```diff
@@ -437,6 +438,43 @@ func TestValidateCreate(t *testing.T) {
...
+					Image: &Image{
+						URL: "test1",
+					},
+				},
+			},
+			oldBMH:    nil,
+			wantedErr: "Image URL test1 is an invalid URL",
+		},
+		{
+			name: "liveISOImageWithUnsupportBMC",
```
Comment (furkatgofurov7) [inline apis/metal3.io/v1alpha1/baremetalhost_validation_test.go:460]:
```suggestion
			name: "liveISOImageWithUnsupportedBMC",
```

Comment (zhouhao3) [reply apis/metal3.io/v1alpha1/baremetalhost_validation_test.go]: Done

**Review by furkatgofurov7 — APPROVED** [review]
/lgtm

**Pull request merged**